indico/modules/rb_new/client/js/common/rooms/RoomPhoto.jsx

```diff
@@ -0,0 +1,191 @@
+    const [file, setFile] = useState(null);
+    const [loading, setLoading] = useState(false);
+    const [hasPreview, setHasPreview] = useState(false);
+
+    useEffect(() => {
+        setHasPreview(hasPhoto);
+    }, [hasPhoto]);
+
+    const onDrop = useCallback(acceptedFiles => {
+        setFile(acceptedFiles[0]);
+        setHasPreview(true);
+    }, []);
+
+    const deletePhoto = async () => {
+        setLoading(true);
+        try {
+            await indicoAxios.delete(roomPhotoURL({room_id: roomId}));
+        } catch (e) {
+            handleAxiosError(e);
+        }
+        setFile(null);
+        setLoading(false);
+        setHasPreview(false);
+    };
+
+    const savePhoto = async () => {
+        setLoading(true);
+        try {
+            const bodyFormData = new FormData();
+            bodyFormData.append('photo', file);
+            const config = {
+                headers: {'content-type': 'multipart/form-data'}
+            };
+            await indicoAxios.post(roomPhotoURL({room_id: roomId}), bodyFormData, config);
+        } catch (e) {
+            handleAxiosError(e);
+        }
+        setFile(null);
+        setLoading(false);
+        setHasPreview(true);
+    };
+
+    const cancelPhoto = () => {
+        setFile(null);
+        setHasPreview(hasPhoto);
+    };
+
+    const getPreview = () => {
+        const randomId = new Date().getTime();
+        return file ? URL.createObjectURL(file) : `${roomPhotoURL({room_id: roomId})}?random=${randomId}`;
+    };
+
+    const dropzoneRef = useRef();
+
+    const openDialog = () => {
+        // Note that the ref is set async, so it might be null at some point
+        if (dropzoneRef.current) {
+            dropzoneRef.current.open();
+        }
+    };
+
+    const uploadBtn = (
+        <Button disabled={loading} type="button" styleName="upload-icon" size="small" onClick={openDialog}>
+            <Icon name="upload" /><Translate>Upload</Translate>
+        </Button>
+    );
+
+    const saveBtn = (
+        <Button disabled={loading || !file}
+                type="button"
+                styleName="save-icon"
+                size="small"
+                onClick={savePhoto}>
+            <Icon name="check" /><Translate>Save</Translate>
+        </Button>
+    );
+
+    const deleteBtn = (
+        <Button disabled={!hasPreview || loading}
+                type="button"
+                styleName="remove-icon"
+                size="small"
+                onClick={deletePhoto}>
+            <Icon name="trash" /><Translate>Delete</Translate>
+        </Button>
+    );
+
+    const cancelBtn = (
+        <Button disabled={loading}
+                type="button"
+                size="small"
+                onClick={cancelPhoto}>
+            <Icon name="dont" /><Translate>Cancel</Translate>
+        </Button>
+    );
+
+    const buttons = (
+        <div styleName="photo-actions">
+            {!file && (
```

Comment:
I think this would be much more readable:

```jsx
const buttons = (
    <div styleName="photo-actions">
        {file ? (
            <>
                <Popup trigger={saveBtn}
                       content={Translate.string('Save photo')} />
                <Popup trigger={cancelBtn}
                       content={Translate.string('Cancel photo upload')} />
            </>
        ) : (
            <>
                <Popup trigger={uploadBtn}
                       content={Translate.string('Upload a new photo')} />
                <Popup trigger={deleteBtn}
                       content={Translate.string('Delete existing photo')} />
            </>
        )}
    </div>
);
```